src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs

```diff
@@ -39,16 +40,19 @@ public FhirCosmosClientInitializer(
             ICosmosClientTestProvider testProvider,
             Func<IEnumerable<RequestHandler>> requestHandlerFactory,
             RetryExceptionPolicyFactory retryExceptionPolicyFactory,
+            Lazy<TokenCredential> tokenCredential,
```

Comment:
Does this need to be Lazy? I think the implementations of `TokenCredential` already are lazy and the wait is when the token is requested.

Reply:
...good point, I was thinking to not resolve it at all if the key is still being used. Not sure I have a strong opinion on this